app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html

```diff
@@ -59,63 +59,128 @@
-        Git Repository URL <help-field class="help-field-absolute" key="appengine.serverGroup.repositoryUrl"></help-field>
+    <div class="form-group row">
+      <label class="col-md-3 sm-label-right">Source Repository Type</label>
+      <div class="col-md-9">
+        <select class="input-sm"
+                ng-options="option for option in basicSettingsCtrl.sourceRepositoryTypes"
+                ng-model="command.sourceRepositoryType">
+          <option value="">Select...</option>
+        </select>
       </div>
-      <div class="col-md-7">
-        <input type="text"
-               required
-               class="form-control input-sm"
-               name="gitRepo"
-               ng-model="command.repositoryUrl"/></div>
     </div>
 
-    <div class="form-group">
-      <div class="col-md-3 sm-label-right">
-        Git Credential Type <help-field class="help-field-absolute" key="appengine.serverGroup.gitCredentialType"></help-field>
+    <div ng-if="basicSettingsCtrl.isArtifactSource()">
+      <div class="form-group">
+        <label class=" col-md-3 sm-label-right">Expected Artifact</label>
+        <div class="col-md-7">
+          <ui-select ng-if="command.backingData.expectedArtifacts.length"
+                     ng-model="command.expectedArtifact"
+                     ng-change="basicSettingsCtrl.setFromArtifact()"
+                     class="form-control input-sm">
+            <ui-select-match>{{ basicSettingsCtrl.summarizeExpectedArtifact(command.expectedArtifact) }}
+            </ui-select-match>
+            <ui-select-choices repeat="expected in command.backingData.expectedArtifacts">
+              <span>{{ basicSettingsCtrl.summarizeExpectedArtifact(expected) }}</span>
+            </ui-select-choices>
+          </ui-select>
+        </div>
       </div>
-      <div class="col-md-7">
-        <select class="form-control input-sm"
-                ng-options="basicSettingsCtrl.humanReadableGitCredentialType(type) for type in basicSettingsCtrl.getSupportedGitCredentialTypes()"
-                ng-model="command.gitCredentialType">
-        </select>
+      <div class="form-group">
+        <label class="col-md-3 sm-label-right">Storage Account</label>
+        <div class="col-md-9">
+          <select class="input-sm"
+                  ng-options="option for option in basicSettingsCtrl.appEngineStorageAccounts"
+                  ng-model="command.storageAccountName">
+            <option value="">Select...</option>
+          </select>
+        </div>
       </div>
     </div>
 
-    <div class="form-group">
-      <div class="col-md-3 sm-label-right">
-        Branch <help-field key="appengine.serverGroup.branch"></help-field>
+    <div ng-if="basicSettingsCtrl.isGcsSourceRepo()">
+      <div class="form-group">
+        <div class="col-md-3 sm-label-right">
+          GCS URL <help-field class="help-field-absolute" key="appengine.serverGroup.repositoryUrl"></help-field>
```

Comment:
The help text here (from `appengine.serverGroup.repositoryUrl`) only makes sense for git repos - I would make a separate help field.